pkg/api/universal/conversation_middleware.go

```diff
@@ -0,0 +1,115 @@
+/*
```

Comment:
Please refrain from creating file names with underscore deliminator between words - needing to do so signals that we should organize into different packages.